### PR TITLE
Move ModelConfigValues to Mode Receiver

### PR DESCRIPTION
**Pull request by ExternalReality**
Why is this change needed?

CAAS Refactoring

### CODE REVIEW

File: apiserver/facades/client/modelconfig/backend.go [hunk 1]
```diff
@@ -25,6 +25,11 @@ type Backend interface {
 
 type stateShim struct {
 	*state.State
+	model *state.Model
+}
+
+func (st stateShim) ModelConfigValues() (config.ConfigValues, error) {
```
Comment (wallyworld) [inline apiserver/facades/client/modelconfig/backend.go]: This shouldn't be on stateShim - it should be on modelShim

**Comment by wallyworld** [comment]
After discussing, we can land this and iterate.

**Comment by ExternalReality** [comment]
$$MERGE$$

**Comment by ExternalReality** [comment]
$$MERGE$$

**Pull request merged**